### PR TITLE
Swapd: state refactors

### PR DESCRIPTION
**Pull request by Lederstrumpf**
A bunch of small independent refactors segregated by commits not tracked by open issues. 
The commit 09387540ecce3f9314396ddd29582781a28925b4 (*couple swap role and wallet role when initiating maker*) I'm unsure about, but would like to discuss stronger coupling of swap roles and wallet roles in handling.

### CODE REVIEW

File: src/swapd/swap_state.rs [hunk 1]
```diff
@@ -715,20 +715,31 @@ fn attempt_transition_to_init_maker(
...
-                    local_params,
-                    remote_commit: commit,
-                    wallet,
-                }))),
+            match (swap_role, wallet.clone()) {
+                (SwapRole::Bob, Wallet::Bob(BobState { funding_tx, .. })) => {
+                    Ok(Some(SwapStateMachine::BobInitMaker(BobInitMaker {
+                        local_commit,
+                        local_params,
+                        funding_address: funding_tx.get_address().ok().unwrap(),
```
Comment (TheCharlatan) [inline src/swapd/swap_state.rs:726]: Rather evaluate this error instead.

File: src/swapd/swap_state.rs [hunk 2]
```diff
@@ -715,20 +715,31 @@ fn attempt_transition_to_init_maker(
...
+                (SwapRole::Alice, Wallet::Alice(_)) => {
+                    Ok(Some(SwapStateMachine::AliceInitMaker(AliceInitMaker {
+                        local_params,
+                        remote_commit: commit,
+                        wallet,
+                    })))
+                }
+                _ => {
+                    runtime.log_error(format!(
+                        "Invalid swap role {} for wallet {:?}",
```
Comment (TheCharlatan) [inline src/swapd/swap_state.rs:741]: Implement format strings on Wallet enum, no need to debug display its entire content.

File: src/swapd/swap_state.rs [hunk 3]
```diff
@@ -541,7 +541,10 @@ impl StateMachine<Runtime, Error> for SwapStateMachine {
             }
             SwapStateMachine::AlicePunish => try_alice_punish_to_swap_end(event, runtime),
 
-            _ => Ok(Some(self)),
+            SwapStateMachine::SwapEnd(_) => {
+                runtime.log_warn("Received event after swap end");
```
Comment (TheCharlatan) [inline src/swapd/swap_state.rs:545]: `warn` level seems a bit much. There are bound to be some incoming events sometimes. I'd rather return `None` instead and let the executor log on `debug` level.

File: src/swapd/wallet.rs [hunk 4]
```diff
@@ -752,8 +752,9 @@ impl Wallet {
                     }
                 } else {
                     error!(
-                        "{} | Wallet not found or not on correct state",
+                        "{} | Not correct wallet, should be Bob: {:?}",
```
Comment (TheCharlatan) [inline src/swapd/wallet.rs:755]: Same thing with the default wallet formatter.

**Review by TheCharlatan — CHANGES_REQUESTED** [review]

File: src/swapd/wallet.rs (hunk 4 above)
Comment (Lederstrumpf) [reply src/swapd/wallet.rs:755]: b1469847

File: src/swapd/swap_state.rs (hunk 3 above)
Comment (Lederstrumpf) [reply src/swapd/swap_state.rs]: b1469847

File: src/swapd/wallet.rs (hunk 4 above)
Comment (Lederstrumpf) [reply src/swapd/wallet.rs]: adc8e0ee

File: src/swapd/swap_state.rs (hunk 2 above)
Comment (Lederstrumpf) [reply src/swapd/swap_state.rs]: ddb110cc

File: src/swapd/wallet.rs (hunk 4 above)
Comment (Lederstrumpf) [reply src/swapd/wallet.rs]: ddb110cc

File: src/swapd/swap_state.rs (hunk 1 above)
Comment (Lederstrumpf) [reply src/swapd/swap_state.rs]: 93be0f39

Comment (Lederstrumpf) [reply src/swapd/swap_state.rs]: 7c275efe

**Review by TheCharlatan — APPROVED** [review]

File: src/swapd/wallet.rs [hunk 5]
```diff
@@ -69,9 +69,11 @@ pub struct HandleBuyProcedureSignatureRes {
     pub buy_tx: bitcoin::Transaction,
 }
 
-#[derive(Clone, Debug, StrictEncode, StrictDecode)]
+#[derive(Clone, Display, Debug, StrictEncode, StrictDecode)]
 pub enum Wallet {
+    #[display("Wallet::Alice")]
```
Comment (h4sh3d) [inline src/swapd/wallet.rs:74]: Maybe a more human readable form for display would be good? e.g. `Alice wallet`

File: src/swapd/wallet.rs [hunk 6]
```diff
@@ -752,8 +754,9 @@ impl Wallet {
                     }
                 } else {
                     error!(
-                        "{} | Wallet not found or not on correct state",
+                        "{} | Not correct wallet, should be Wallet::Bob: {}",
```
Comment (h4sh3d) [inline src/swapd/wallet.rs:757]: Same here `Bob's wallet`

File: src/swapd/wallet.rs (hunk 5 above)
Comment (Lederstrumpf) [reply src/swapd/wallet.rs]: 1d1e45bd81f003cfcbec79987954ae7647e79fd5

File: src/swapd/wallet.rs (hunk 6 above)
Comment (Lederstrumpf) [reply src/swapd/wallet.rs]: 1d1e45bd81f003cfcbec79987954ae7647e79fd5

**Review by h4sh3d — APPROVED** [review]

**Pull request merged**